### PR TITLE
[MOB-9997] bug fix user update aggregation and update user rearchitect

### PR DESCRIPTION
**Pull request by evantk91**
## 🔹 Jira Ticket(s) if any

* [MOB-9997](https://iterable.atlassian.net/browse/MOB-9997)
* [MOB-9998](https://iterable.atlassian.net/browse/MOB-9998)

## ✏️ Description

> Please provide a brief description of what this pull request does.

[MOB-9997]: https://iterable.atlassian.net/browse/MOB-9997?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[MOB-9998]: https://iterable.atlassian.net/browse/MOB-9998?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by Ayyanchira — APPROVED** [review]
LGTM. Made nit pick comments

File: iterableapi/src/main/java/com/iterable/iterableapi/AnonymousUserManager.java [hunk 1]
```diff
@@ -358,15 +344,12 @@ private void storeEventListToLocalStorage(JSONObject newDataObject, boolean shou
         if (!iterableApi.getAnonymousUsageTracked()) {
             return;
         }
-        JSONArray previousDataArray = getEventListFromLocalStorage();
-        if (shouldOverWrite) {
-            previousDataArray = removeDuplicateEvent(previousDataArray, newDataObject);
```
[the remark this answers is not in the document]
Comment (evantk91) [reply iterableapi/src/main/java/com/iterable/iterableapi/AnonymousUserManager.java:363]: The purpose was to remove the old user update event which was not correct. We needed to aggregate the user updates together.

**Review by bradumbaugh — COMMENTED** [review]
@evangreer91 Took a quick glance a this — thanks for putting it together. Left you one comment about how arrays should be merged..

File: iterableapi/src/main/java/com/iterable/iterableapi/AnonymousUserManager.java [hunk 2]
```diff
@@ -377,29 +350,51 @@ private void storeEventListToLocalStorage(JSONObject newDataObject, boolean shou
...
+                    // For simple types, overwrite
+                    target.put(key, value);
+                }
+            } else {
+                // If the key doesn't exist in the target, just add it
+                target.put(key, value);
             }
         }
-        return newArray;
     }
```
[the remark this answers is not in the document]
Comment (bradumbaugh) [reply iterableapi/src/main/java/com/iterable/iterableapi/AnonymousUserManager.java:408]: @evantk91 My other question here is whether we're ready to go down the path of pulling the user profile updates outside of the event updates. I think it's the right thing to do, but doing it here before beta implies that we'll do it in the other SDKs too, I think...

[the remark this answers is not in the document]
Comment (evantk91) [reply iterableapi/src/main/java/com/iterable/iterableapi/AnonymousUserManager.java:408]: Lets ask the team

[the remark this answers is not in the document]
Comment (evantk91) [reply iterableapi/src/main/java/com/iterable/iterableapi/AnonymousUserManager.java:408]: I will merge this and cut a separate ticket if we want to change this behavior.

[the remark this answers is not in the document]
Comment (evantk91) [reply iterableapi/src/main/java/com/iterable/iterableapi/AnonymousUserManager.java:408]: It is already being done on the other SDKs

**Pull request merged**